doc/source/testing_pull_requests.rst

```diff
@@ -0,0 +1,178 @@
+Testing an python-for-android pull request
+==========================================
+
+In order to test a p4a pull requests we contemplate three possible scenarios:
+
+  - using python-for-android commands directly from the pull request files
+    that we want to test, without installing it (the recommended way for most
+    of the test cases)
+  - installing python-for-android using the github's branch of the pull request
+  - using buildozer and a custom app
+
+We will explain the first two methods using one of the distributed
+python-for-android test apps and we assume that you already have the
+python-for-android dependencies installed. For the `buildozer` method we assume that you
+already have a a properly working app to test and a working
+installation/configuration of `buildozer`. There is one step that it's shared
+with all the testing methods that we propose in here...we named it
+`Common steps`.
+
+
+Common steps
+^^^^^^^^^^^^
+
+- open a terminal, create a test directory and enter inside it, eg.::
+
+    .. codeblock:: bash
+
+        mkdir p4a_pull_requests
+        cd p4a_pull_requests
+
+- clone the pull request you wanna to test, let's imagine that you wanna to
+  test a pull request whose branch is named `feature-fix-numpy` and the github
+  nickname of the user who created the pull request is named `obiwankenoby`::
+
+    .. codeblock:: bash
+
+        git clone -b feature-fix-numpy \
+            --single-branch \
+            https://github.com/obiwankenoby/python-for-android.git \
+            p4a-feature-fix-numpy
+
+For what it's worth it about the command above explained line by line:
+
+    - `git clone -b feature-fix-numpy`: we tell git that we want to clone the
+      branch named `feature-fix-numpy`
+    - `--single-branch`: we tell git that we only want that branch
+    - `https://github.com/obiwankenoby/python-for-android.git`: noticed the
+      nickname of the user that created the pull request: `obiwankenoby` in the
+      middle of the line? that should be changed as needed for each pull
+      request that you wanna to test
+    - `p4a-feature-fix-numpy`: the name of the cloned repository, so we can
+      have multiple clones of different prs in the same folder
```

Comment:
I actually assumed this part is known by contributors, so I would also have skipped that part. But it's good that you have it there then.
There's also a "shortcut" to actually fetch the PR given its number:
```sh
git fetch origin pull/<#>/head:<local_branch_name
```
For instance for yours:
```
git fetch upstream pull/1901/head:feature-docs-test-pr
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 6 (delta 4), reused 6 (delta 4), pack-reused 0
Unpacking objects: 100% (6/6), done.
From https://github.com/kivy/python-for-android
 * [new ref]           refs/pull/1901/head -> feature-docs-test-pr
[git checkout feature-docs-test-pr
Switched to branch 'feature-docs-test-pr'
```
The workflow I personally use most of the time is to add the remote and checkout the branch. So for yours:
```sh
git remote add opacam https://github.com/opacam/python-for-android.git
git fetch opacam
git checkout opacam/feature-docs-test-pr
```

Reply:
Oh yeah, I will write that :wink: